src/components/DeviceWidth/index.js

```diff
@@ -0,0 +1,36 @@
+import {Component} from 'react'
+import {mobileMediaQuerySize} from 'utils/hardCodedSizes'
+
+export default class extends Component {
+  
+  state = {
+    isMobile: false,
+  }
+
+  componentDidMount = () => {
+    this.handleWidthChange()
+    window.onresize = this.handleWidthChange
+  }
+
+  handleWidthChange = () => {
+    const {onWidthChange} = this.props
+    if(onWidthChange) {
+      onWidthChange()
+    }
+    if(window.screen.width < mobileMediaQuerySize) {
+      this.setState({
+        isMobile: true,
+      })
+    } else {
+      this.setState({
+        isMobile: false,
+      })
+    }
+  }
+
+  render() {
+    const {isMobile} = this.state
+    const {children} = this.props
+    return children(isMobile)
+  }
+}
```

Comment:
Thanks :) I love Render Callbacks (AKA function as child); makes it really easy to share stateful logic across components. Here is my snippet for them if interested:
https://github.com/trevordmiller/settings/blob/master/syncfiles/UltiSnips/all.snippets#L119-L134